azext_iot/deviceupdate/commands_instance.py

```diff
@@ -0,0 +1,120 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azext_iot.common.utility import handle_service_exception
+from azext_iot.deviceupdate.providers.base import (
+    DeviceUpdateMgmtModels,
+    DeviceUpdateInstanceManager,
+    parse_account_rg,
+)
+
+logger = get_logger(__name__)
+
+# Instances
+
+
+def create_instance(
+    cmd,
+    name,
+    instance_name,
+    iothub_resource_ids,
+    diagnostics=None,
+    storage_resource_id=None,
+    location=None,
+    tags=None,
+    resource_group_name=None,
+):
+    instance_manager = DeviceUpdateInstanceManager(cmd=cmd)
```

Comment:
Is there a reason for having this kind of command structure (similar to how the dps.py or iothub.py in operations is structured) rather than the structure of having providers (iothub/providers/job, pnp_runtime, etc)

Reply:
The desire here, having experienced the multiple command structure patterns, is a balanced approach. The ADU CLI carves a new independent command space with self contained command definitions, params, help and providers just like the digital twins CLI and other newer CLI areas. The goal for abstraction is to provide value vs abstraction for the sake of abstraction.

This PR shows a work-in-progress state, feature complete on control plane, however significant chunks of the ADU CLI still need to be developed, and current patterns further fleshed out.